scantool/diag.h

```diff
@@ -334,6 +336,31 @@ char **strlist_add(char **list, const char *news, int elems);
  */
 void strlist_free(char **slist, int elems);
 
+// Atomics
+
+// Atomically accessed types
+typedef struct {
```

Comment:
Why is this not an opaque format kept internally within diag_os.c ? code including diag.h really shouldn't see this struct